fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeMetadata.java

```diff
@@ -0,0 +1,62 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.connector.delta;
+
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.HiveTable;
+import com.starrocks.catalog.Table;
+import com.starrocks.connector.ConnectorMetadata;
+import com.starrocks.external.delta.DeltaUtils;
+import com.starrocks.external.hive.HiveMetastoreOperations;
+import io.delta.standalone.DeltaLog;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.metastore.conf.MetastoreConf;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+
+public class DeltaLakeMetadata implements ConnectorMetadata  {
+    private static final Logger LOG = LogManager.getLogger(DeltaLakeMetadata.class);
+    private final String resourceName;
+    private final String catalogName;
+    private final HiveMetastoreOperations hmsOps;
+
+    public DeltaLakeMetadata(String resourceName, String catalogName, HiveMetastoreOperations hmsOps) {
+        this.resourceName = resourceName;
+        this.catalogName = catalogName;
+        this.hmsOps = hmsOps;
+    }
+
+    @Override
+    public List<String> listDbNames() {
+        return hmsOps.getAllDatabaseNames();
+    }
+
+    @Override
+    public List<String> listTableNames(String dbName) {
+        return hmsOps.getAllTableNames(dbName);
+    }
+
+    @Override
+    public Database getDb(String dbName) {
+        return hmsOps.getDb(dbName);
+    }
+
+    @Override
+    public List<String> listPartitionNames(String databaseName, String tableName) {
+        return hmsOps.getPartitionKeys(databaseName, tableName);
+    }
+
+    @Override
+    public Table getTable(String dbName, String tblName) {
+        Table table = hmsOps.getTable(dbName, tblName);
+        HiveTable hiveTable = (HiveTable) table;
+        String path = hiveTable.getHdfsPath();
+        Configuration configuration = new Configuration();
+        configuration.set(MetastoreConf.ConfVars.THRIFT_URIS.getHiveName(), resourceName);
+        DeltaLog deltaLog = DeltaLog.forTable(configuration, path);
```

Comment:
I think DeltaLog may be managed and initialized in DeltaLakeTable. There is no need to construct the configuration and delta log every time.